### PR TITLE
Port `find` command

### PR DESCRIPTION
**Pull request by arthur-targaryen**
- Implement `find` command (#242).
- Add `Value::test_nothing()`.

### CODE REVIEW

**Comment by fdncred** [comment]
looking good so far. looks like it's going to be an upgrade to the nushell find command.

**Comment by arthur-targaryen** [comment]
Thanks! What has been done with the `PipelineIterator` is great and helps a lot.

**Comment by fdncred** [comment]
How's it going on this pr @arthur-targaryen? I remember having some discussion on this topic on discord.

**Comment by arthur-targaryen** [comment]
> How's it going on this pr @arthur-targaryen? I remember having some discussion on this topic on discord.

I didn't find the time to work on this this week but I've planned to finish it tomorrow 🙂

**Comment by fdncred** [comment]
Sounds good. Thanks for the help.

File: crates/nu-command/src/filters/find.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                            Value::String { .. }
+                            | Value::List { .. }
+                            | Value::CustomValue { .. } => term
+                                .r#in(span, value)
+                                .map_or(false, |value| value.is_true()),
+                            Value::Record { vals, .. } => vals.iter().any(|val| {
+                                term.r#in(span, val).map_or(false, |value| value.is_true())
+                            }),
+                            Value::Binary { .. } => todo!(),
+                            Value::CellPath { .. } => todo!(),
```
Comment (arthur-targaryen) [inline crates/nu-command/src/filters/find.rs:146]: Not sure how those should be handled.

Comment (fdncred) [reply crates/nu-command/src/filters/find.rs:158]: I'm not sure. I'd have to look to see how nushell handles them. Was there anything useful there when you looked?

Comment (arthur-targaryen) [reply crates/nu-command/src/filters/find.rs]: In nushell `find` converts each element of the input stream to a trimmed, lowercase string using `convert_to_string().trim().to_lowercase()`. Then it uses `str::contains` with the rest parameters of the command.

I tried to mimic this behavior for `Value::CellPath` in engine-q (see 45c006f).
However, the `convert_to_string()` method on `Primivite::Binary` in nushell always return an empty string, so we need to decide if we choose to change it, and if so how should it be changed.

Comment (jntrnr) [reply crates/nu-command/src/filters/find.rs]: @arthur-targaryen - I don't think we support it just yet, but if you want, I think it'd be fine to file an issue and go with the assumption that we'll add support in the future. It shouldn't directly impact the find work.

**Comment by jntrnr** [comment]
@arthur-targaryen - is this ready? or are there things missing that need to be added before it's ready?

**Comment by arthur-targaryen** [comment]
> @arthur-targaryen - is this ready? or are there things missing that need to be added before it's ready?

Following you reply to my comment, I just want to replace the `todo!` with a proper error and it should be ready to merge!

**Comment by kubouch** [comment]
@arthur-targaryen Awesome! Can you also clear out the merge conflicts?

**Comment by fdncred** [comment]
@arthur-targaryen I'll land this if we're ready. Are we?

**Comment by arthur-targaryen** [comment]
> @arthur-targaryen I'll land this if we're ready. Are we?

Yep we are, finally 🙂

**Comment by fdncred** [comment]
Thanks for all your hard work on this!!!

**Pull request merged**